modal/lerobot_modal/conversion_service.py

```diff
@@ -0,0 +1,95 @@
+import modal
+
+
+conversion_image = (
+    modal.Image.debian_slim(python_version="3.10")
+    .apt_install(
+        "ffmpeg",
+        "libavutil-dev",
+        "libavcodec-dev",
+        "libavformat-dev",
+    )
+    .env(
+        {
+            "HF_HUB_ENABLE_HF_TRANSFER": "1",
+            "HF_HUB_DISABLE_TELEMETRY": "1",
+        }
+    )
+    .uv_pip_install(
+        "lerobot==0.4.0",  # This version includes v3.0 of LeRobotDataset
+        "loguru",
+        "huggingface_hub[cli]",
+    )
+)
+
+conversion_service = modal.App("conversion-app")
+
+
+@conversion_service.function(
+    image=conversion_image,
+    cpu=1,  # Single core is enough from experience
+    timeout=15 * 60,
+    secrets=[
+        modal.Secret.from_dict({"MODAL_LOGLEVEL": "DEBUG"}),
+        modal.Secret.from_name("huggingface"),
+    ],
+)
+async def convert_dataset_to_v3(
+    dataset_name: str,
+    huggingface_token: str | None = None,
+) -> tuple[str | None, str | None]:
+    """
+    Convert a v2.1 dataset to LeRobot v3.0 format and upload to Hugging Face Hub.
+    If a token is passed, we assume it has the necessary permissions to push to the dataset.
+    If not, we reupload a version of the dataset on the phospho cloud.
+    """
+    import os
+    from loguru import logger
+    from huggingface_hub import (
+        snapshot_download,
+        upload_folder,
+        create_repo,
+        create_tag,
+    )
+    from lerobot.datasets.v30.convert_dataset_v21_to_v30 import convert_dataset
+
+    try:
+        # Clear Hugging Face cache
+        os.system("huggingface-cli delete-cache -y")
```

Comment:
Not sure why you'd do this? On the contrary, we'd want to use the HF cache that is being populated by any previous pipeline. 
Eg: 
1. Train ACT
2. Downloads HF dataset to cache (modal volume)
3. Detects that it's v2.1
4. Call this microservice "v3 conversion"
5. Downloads HF dataset from the cache (modal volume)
6. Push new version of the dataset